lib/trento/application/integration/discovery/payloads/sap_system_discovery_payload.ex

```diff
@@ -0,0 +1,349 @@
+defmodule Trento.Integration.Discovery.SapSystemDiscoveryPayload do
+  @moduledoc """
+  SAP system discovery integration event payload
+  """
+
+  alias Trento.Integration.Discovery.SapSystemDiscoveryPayload.{
+    Database,
+    Instance,
+    Profile
+  }
+
+  @required_fields [:Id, :SID, :Type, :Profile, :Databases, :Instances]
+
+  use Trento.Type
+
+  deftype do
+    field :Id, :string
```

Comment:
AFAIK, fields should be starting with a lowercase.

Is this related to your comment 
_I couldn't use the ProperCase option to snake case all the keys, as it creates some really strange entries_ ?

If so I couldn't fully understand what is happening :sweat_smile:

Reply:
This fields are exactly what we receive in the discovery json payload. In this case we receive the `Id`, as it is, starting upper case.
In other piece of code, @dottorblaster used the `ProperCase.snake_case` function to convert the coming payload to snake-cased payload, so he could have snake case fields. In essence, the payload schemas he created doesn't 100% match with the coming payload, as he is pre-snake-casing (nice word XD).

The reason why I didn't use the `ProperCase.snake_case` function is that it creates really strange keys, because our coming payload has strange keys indeed. Many of them because we simply send but SAP commands give us...

Reply:
@arbulu89 
could you please make an example of what are this strange keys? maybe I would be able to help with this or even rewrite our version of proper case that fits for our usecase (including the "/" keys)

Reply:
Here some few:
- `HdbnsutilSRstate` to `hdbnsutil_s_rstate`
- `icm/HTTP/ASJava/disable_url_session_tracking` to `icm/_htt_p/_as_java/disable_url_session_tracking`
- `is/HTTP/show_detailed_errors` to `is/_htt_p/show_detailed_errors`
- `siteTier/NBG` to `site_tier/_nbg`

From this list, we really only use the 1st one. I didn't want to start having this strange fields in the code.
I thought of changing the agent side too, but I didn't want to start on this until having the payload versioning system on place.

If you don't mind having these fields, we can snake case anyway

Reply:
@arbulu89 
 I see thanks. I started playing around but I ended up with the same problem as the propercase library. It uses Macro.underscore under the hood which causes the "weird" casing. We will want to change the agent side in the future I agree. Since there is no easy way to do this, and what you proposed is limited to the payload decoded (and not to the commands itself) I would think this is good enough (even if I cringe everytime I see a camlecase atom field 😬 )

Reply:
We can have poll hehe;

Who prefers the:
- ugly CamelCase entries
- ugly snake_case entries

Either way, there will be strange fields.